framework/source/class/qx/util/DynamicScriptLoader.js

```diff
@@ -143,24 +143,36 @@ qx.Class.define("qx.util.DynamicScriptLoader", {
 
     /**
      * Start loading scripts. This may only be called once!
-     * @return {Promise} a promise which will be resolved after load of all scripts.
+     * @return {Promise?} a promise which will be resolved after load of all scripts if promises are enabled.
```

Comment:
Shouldn't it document what it returns if promises are NOT enabled? Sp that the API consumer knows how to code against this?

Reply:
But it doesn't return anything if promises are NOT enabled. It only returns something (a promise) if promises are enabled and that is documented.

Reply:
Since it was unclear to one person, it may be unclear to others, too. Maybe change the documentation for that return value to,
```
a promise which will be resolved after load of all scripts if promises are enabled, if promise support is enabled; nothing (undefined) if promises are not enabled.
```

Reply:
I don't insist - it wasn't clear to me that the JSDOC "{Promise?}" means "{Promise|undefined}" but if it does, that should be enough.

Reply:
Well I think @derrel is right: it is better to mention the `undefined` return explicitly.